mlx/backend/accelerate/primitives.cpp

```diff
@@ -193,6 +193,25 @@ void ArcTan::eval_cpu(const std::vector<array>& inputs, array& out) {
   }
 }
 
+void ArcTan2::eval_cpu(const std::vector<array>& inputs, array& out) {
+  assert(inputs.size() == 2);
+  auto& a = inputs[0];
+  auto& b = inputs[1];
+  if (out.dtype() == float32 && a.flags().contiguous && b.flags().contiguous) {
+    if (a.is_donatable() && a.itemsize() == out.itemsize()) {
+      out.copy_shared_buffer(a);
+    } else if (b.is_donatable() && b.itemsize() == out.itemsize()) {
+      out.copy_shared_buffer(b);
+    } else {
+      out.set_data(allocator::malloc_or_wait(out.nbytes()));
+    }
+    int size = a.data_size();
+    vvatan2f(out.data<float>(), a.data<float>(), b.data<float>(), &size);
+  } else {
+    eval(inputs, out);
+  }
```

Comment:
This is not correct. Could you follow the example of `Add` (or any other binary primitive)?

Reply:
Ah, gotcha. I was following `Power`. I'll update it now :)

Reply:
Actually following `Power` might be better given there are no scalar equivalents. However, you would need to use `row_contiguous` here and not `contiguous` (which is a bug).

Reply:
Ah, I was stuck on that and just about to ask you :) I'll re-run tests/lint and push the commit to the PR.